### ISSUE TITLE
/pg_prometheus.so: undefined symbol: PG_GETARG_JSONB"

### ISSUE THREAD

**Issue by Rajinimaybank** [timescale/prometheus-postgresql-adapter#54]
./prometheus-postgresql-adapter -log.level=debug -leader-election.pg-advisory-lock-id=1 -leader-election.pg-advisory-lock.prometheus-timeout=6s -pg.password=root -pg.use-timescaledb=false
level=info ts=2018-11-27T09:04:28.2253517Z caller=log.go:25 config="&{remoteTimeout:30000000000 listenAddr::9201 telemetryPath:/metrics pgPrometheusConfig:{host:localhost port:5432 user:postgres password:root database:postgres schema: sslMode:disable table:metrics copyTable: maxOpenConns:50 maxIdleConns:10 pgPrometheusNormalize:true pgPrometheusLogSamples:false pgPrometheusChunkInterval:43200000000000 useTimescaleDb:false dbConnectRetries:0 readOnly:false} logLevel:debug haGroupLockId:1 restElection:false prometheusTimeout:6000000000}"
level=info ts=2018-11-27T09:04:28.2305317Z caller=log.go:25 msg="host=localhost port=5432 user=postgres dbname=postgres password='root' sslmode=disable connect_timeout=10"
level=error ts=2018-11-27T09:04:28.2873623Z caller=log.go:33 err="pq: could not load library \"/usr/lib/postgresql/11/lib/pg_prometheus.so\": /usr/lib/postgresql/11/lib/pg_prometheus.so: undefined symbol: PG_GETARG_JSONB"

Can some one help me on the same

**Comment by niksajakovljevic** [comment]
Hi, 
Have you installed https://github.com/timescale/pg_prometheus?
Even if you don't use TimescaleDB, you still need to install `pg_prometheus` extension

**Comment by did-g** [comment]
It seems you have compiled pg_promethus with postgres V11, but PG_RETURN_JSONB_P has been renamed in 
```
commit 4bd1994650fddf49e717e35f1930d62208845974
Author: Tom Lane <tgl@sss.pgh.pa.us>
Date:   Mon Sep 18 15:21:23 2017 -0400
    Make DatumGetFoo/PG_GETARG_FOO/PG_RETURN_FOO macro names more consistent.

#define PG_RETURN_JSONB(x) PG_RETURN_JSONB_P(x)

```
No idea if it's the only issue with newer PG

**Comment by niksajakovljevic** [comment]
@did-g Actually we don't provide `pg_prometheus` binary. You need to build it yourself or use docker.

**Comment by ryanbreed** [comment]
> It seems you have compiled pg_promethus with postgres V11, but PG_RETURN_JSONB_P has been renamed in
> 
> ```
> commit 4bd1994650fddf49e717e35f1930d62208845974
> Author: Tom Lane <tgl@sss.pgh.pa.us>
> Date:   Mon Sep 18 15:21:23 2017 -0400
>     Make DatumGetFoo/PG_GETARG_FOO/PG_RETURN_FOO macro names more consistent.
> 
> #define PG_RETURN_JSONB(x) PG_RETURN_JSONB_P(x)
> ```
> No idea if it's the only issue with newer PG

+1 ran into the same issue with pgdg11 on ubuntu 18.04; just manually hacked src/prom.c:407
`
  Jsonb    *jb = PG_GETARG_JSONB_P(3);
`
re-ran make install and it's at least running; this should be moved over to pg_prometheus to macro/config around it

**Comment by spolcyn** [comment]
Issue was fixed with pg_prometheus PR#36.
